backend/dataall/modules/s3_datasets/api/dataset/resolvers.py

```diff
@@ -156,6 +156,11 @@ def list_datasets_owned_by_env_group(
     return DatasetService.list_datasets_owned_by_env_group(environmentUri, groupUri, filter)
 
 
+def list_object_keys(context, source, datasetUri: str = None):
+    with context.engine.scoped_session() as session:
+        return DatasetService.list_object_keys(session=session, uri=datasetUri)
+
```

Comment:
instead of using the context here an initializing the session in the resolver,  we should not pass the session and use get_context to start the session directly in the service after the permissions are checked

Reply:
yup oversight - fixing now